pkg/controller/dependences.go

```diff
@@ -79,6 +79,10 @@ type CLIConfig struct {
 	WaitDuration          time.Duration
 	// ResyncDuration is the resync time of informer
 	ResyncDuration time.Duration
+	// DetectNodeFailure enables detection of node failures for stateful failure pods for recovery
+	DetectNodeFailure bool
```

Comment:
Please correct my comments if my understanding is incorrect. Maybe rename it to `RecoverNodeFailure`? Detect sounds like a "read-only" operation where we're actually actively trying to recover the node failure by deleting pods.
```suggestion
	// RecoverNodeFailure detects node failures for stateful components and actively attempts to recover by deleting pods
	RecoverNodeFailure bool
```

Reply:
The failure recovery is broken down into two parts -
1. DetectNodeFailure
2. RecoverNodeFailure

DetectNodeFailure flag does the following things - 
1. It checks the k8s Node and Pod Conditions to decide whether the host is down.
2. And based on that it marks the FailureStore or FailureMember as "HostDown".

The actual recovery by deleting pods is controlled by the "auto-failure-recovery" annotation in TidbCluster which is only activated if the "HostDown" is marked true by the DetectNodeFailure part.

Reply:
Ok